Description: QDoc: Fix handling of -isystem option
Author: Dmitry Shachnev <mitya57@debian.org>
Forwarded: https://codereview.qt-project.org/c/qt/qttools/+/292544
Last-Update: 2020-03-01

--- a/src/qdoc/clangcodeparser.cpp
+++ b/src/qdoc/clangcodeparser.cpp
@@ -1059,6 +1059,9 @@ void ClangCodeParser::initializeParser(c
         if (p.startsWith(QLatin1String("-I")) || p.startsWith(QLatin1String("-F"))) {
             rawpath = p.mid(2).trimmed();
             option = p.left(2).toUtf8();
+        } else if (p.startsWith(QLatin1String("-isystem"))) {
+            rawpath = p.mid(8).trimmed();
+            option = "-isystem";
         } else {
             rawpath = p;
             option = "-I";
